### PR TITLE
Colour service integration

### PR DESCRIPTION
**Pull request by Pallemann**
Colour service integration and various small improvements to the GUI.

### CODE REVIEW

**Review by DrJoLe — COMMENTED** [review]
Looks good!

I don't know, if this is fixed somewhere else, but can we shorten the song title, if it gets to large? @Pallemann @exp625 

Capture

**Comment by exp625** [comment]
> 
> 
> Looks good!
> 
> I don't know, if this is fixed somewhere else, but can we shorten the song title, if it gets to large? @Pallemann @exp625
> 
> Capture

Maybe implement this https://css-tricks.com/flexbox-truncated-text/ in combination with https://stackoverflow.com/questions/58749590/animate-scroll-of-text-when-there-is-text-overflow-in-angular-8

File: src/app/core/services/notification.service.ts [hunk 1]
```diff
@@ -15,8 +15,8 @@ export class Notification {
     message: string | TemplateRef<any>,
   ) {
     this.message = message;
-    this.autohide = false;
-    this.delay = 0;
+    this.autohide = true;
```
Comment (exp625) [inline src/app/core/services/notification.service.ts:18]: Either use the setAutohide https://github.com/JamFactoryApp/jamfactory-web/blob/e56023d8b37830b67ddf9221c32051b896679f35/src/app/core/services/notification.service.ts#L37 function to accomplish the autohide behavior or invert the logic everywhere and add a removeAutohide function for notifications that should stay until the user closes them.

**Review by exp625 — CHANGES_REQUESTED** [review]
Looks good! See comment on changing the notification behaviour. Maybe implement the imgStore in the [localStorage](https://developer.mozilla.org/de/docs/Web/API/Window/localStorage) to keep the data on reload.

Notes: 
- On smaller screens there is a white border between the search component and the first Queue Item.
- Currently the whole queue flashes white when the queue (e.g. when voting) is changed. This is probably an issue on its own because optimizing the ngFor is quite complicated...

File: src/app/core/services/notification.service.ts (hunk 1 above)
Comment (Pallemann) [reply src/app/core/services/notification.service.ts:18]: Is there a scenario where the notification should always be shown and not disappear after a given time?

Comment (exp625) [reply src/app/core/services/notification.service.ts:18]: - Notification that the JamSession was closed due to inactivity and
- Notification to indicate that the user should select a playback device 
are two examples that are already implemented

You can also use
https://github.com/JamFactoryApp/jamfactory-web/blob/e56023d8b37830b67ddf9221c32051b896679f35/src/app/core/services/notification.service.ts#L72 to clear all notifications, https://github.com/JamFactoryApp/jamfactory-web/blob/e56023d8b37830b67ddf9221c32051b896679f35/src/app/core/services/notification.service.ts#L80 to clear all notifications without autohide or https://github.com/JamFactoryApp/jamfactory-web/blob/e56023d8b37830b67ddf9221c32051b896679f35/src/app/core/services/notification.service.ts#L76 to delete a specific notification

Comment (DrJoLe) [reply src/app/core/services/notification.service.ts:18]: I think it's better to increase the duration for which these notifications are visible and not make them stay there forever.

Another solution, as a compromise, would be to implement a notification box that can be shown when clicking a button and which contains unseen notifications as well as a badge indicating how many notifications have not been seen by the user yet.

On the other hand this seems like overkill in contrast to the amount of notifications that we will show to the user.

**Comment by DrJoLe** [comment]
In general, I think we should find a better workflow for the future. I like the discussion we have in this PR and would like to propose that we start opening issues, work on them in separate branches and make PRs that will be reviewed by someone else.

- The PRs should contain not much code and thus be easy to review.
- The PRs should ideally not require code changes in the same files to prevent merge conflicts.
- Issues should also be small and consists of a self-contained problem.


What do you guys think? @exp625 @Pallemann 

Also, the discussion in this PR does not really belong to the PR but another issue. This should be discussed in a new issue and this PR should be merged.

File: src/app/core/services/notification.service.ts (hunk 1 above)
Comment (exp625) [reply src/app/core/services/notification.service.ts:18]: I've opend a new issue. @Pallemann please remove the changes to the notification behaviour and resume the discussion in the issue.

**Comment by exp625** [comment]
Yes, this is a good idea. We need a naming convention for the branch names then!

**Comment by exp625** [comment]
@Pallemann could you briefly state if you can easily solve the problem of the white flashing when voting. If not we should merge and create a new Issue.

**Comment by DrJoLe** [comment]
> 
> 
> Yes, this is a good idea. We need a naming convention for the branch names then!

+1

I propose

`123-the-problem-to-solve`

where `123` is the issue's number and `the-problem-to-solve` is the issue's title in lowercase concatenated by hyphens.

**Comment by Pallemann** [comment]
@exp625 working on it. It will hopefully be fixed soon and then it's ready to be merged.

**Comment by DrJoLe** [comment]
Also, this applies to the frontend as well as the backend. For the secret private repository in this organization i would like to postpone this procedure since the major part of that repository will be developed using pair programming.

**Review by exp625 — APPROVED** [review]

**Review by DrJoLe — APPROVED** [review]

**Pull request merged**